### PR TITLE
grant CANCELLER role to deployer wallet on contract deployment [LiFiTimelockController v1.0.1]

### PR DESCRIPTION
**Pull request by 0xDEnYO**
# Which Jira task belongs to this PR?
https://lifi.atlassian.net/browse/LF-14390

# Why did I implement it this way?

# Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] This pull request is as small as possible and only tackles one problem
- [ ] I have added tests that cover the functionality / test the bug
- [ ] For new facets: I have checked all points from this list: https://www.notion.so/lifi/New-Facet-Contract-Checklist-157f0ff14ac78095a2b8f999d655622e
- [ ] I have updated any required documentation

# Checklist for reviewer (DO NOT DEPLOY and contracts BEFORE CHECKING THIS!!!)

- [ ] I have checked that any arbitrary calls to external contracts are validated and or restricted
- [ ] I have checked that any privileged calls (i.e. storage modifications) are validated and or restricted
- [ ] I have ensured that any new contracts have had AT A MINIMUM 1 preliminary audit conducted on <date> by <company/auditor>

### CODE REVIEW

File: src/Security/LiFiTimelockController.sol [hunk 1]
```diff
@@ -26,15 +27,31 @@
...
     ) TimelockController(_minDelay, _proposers, _executors, _admin) {
+        // validate constructor parameters
+        if (
+            _minDelay == 0 ||
+            _proposers.length == 0 ||
+            _executors.length == 0 ||
+            _admin == address(0) ||
+            _diamond == address(0)
+        ) revert InvalidConfig();
+
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply src/Security/LiFiTimelockController.sol]: added

File: test/solidity/Security/LiFiTimelockController.t.sol [hunk 2]
```diff
@@ -66,6 +69,88 @@ contract LiFiTimelockControllerTest is Test {
...
+        // Then deploy timelock with correct mockDiamond address
+        timelock = new LiFiTimelockController(
+            0,
+            proposers,
+            executors,
+            deployerWallet,
+            admin,
+            address(mockDiamond)
+        );
+
```
Comment (mirooon) [inline test/solidity/Security/LiFiTimelockController.t.sol:104]: there is one missing test case for empty proposers array

File: src/Security/LiFiTimelockController.sol [hunk 3]
```diff
@@ -26,15 +27,32 @@ contract LiFiTimelockController is TimelockController {
...
     /// @param _proposers Accounts to be granted proposer and canceller roles
     /// @param _executors Accounts to be granted executor role
+    /// @param _deployerWallet Address of the deployer wallet (that will be granted CANCELLER role)
     /// @param _admin The address that will be the admin of the TimelockController (= the LI.FI MultiSig SAFE)
+    /// @param _diamond The address of the diamond contract that this timelock controls
     constructor(
         uint256 _minDelay,
         address[] memory _proposers,
         address[] memory _executors,
+        address _deployerWallet,
```
Comment (mirooon) [inline src/Security/LiFiTimelockController.sol]: can we name it '_canceller' or '_cancellerWallet'?

File: test/solidity/Security/LiFiTimelockController.t.sol (hunk 2 above)
Comment (0xDEnYO) [reply test/solidity/Security/LiFiTimelockController.t.sol:104]: added

File: src/Security/LiFiTimelockController.sol (hunk 3 above)
Comment (0xDEnYO) [reply src/Security/LiFiTimelockController.sol]: good point, changed.
I also added docs and deploy scripts (they were still in another open PR)

File: script/deploy/facets/DeployLiFiTimelockController.s.sol [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+        address diamondAddress = _getConfigContractAddress(
+            path,
+            ".LiFiDiamond"
+        );
+
+        // get minDelay from config file
+        string memory timelockConfigPath = string.concat(
+            root,
+            "/config/timelockController.json"
```
Comment (mirooon) [inline script/deploy/facets/DeployLiFiTimelockController.s.sol:42]: `timelockController.json` but the config file is named `timelockcontroller.json`

**Review by mirooon — APPROVED** [review]

**Review by mirooon — APPROVED** [review]
